filters/NormalFilter.cpp

```diff
@@ -119,9 +123,20 @@ void NormalFilter::prepared(PointTableRef table)
         m_args->m_up = false;
     }
 
-    // The query point is returned as a neighbor of itself, so we must increase
-    // k by one to get the desired number of neighbors.
-    ++m_args->m_knn;
+    if (m_radiusArg->set() && m_args->m_knn)
+    {
```

Comment:
This seems like a confusing warning. It's one or the other (radius or count). If someone sets both, I would think that's a hard error because they don't know what they're doing. Just going on seems an invitation to trouble.

Reply:
You already have code that is checking if an argument is set...